src/System.Runtime/tests/System/ExceptionTests.cs

```diff
@@ -208,16 +208,31 @@ private static void ThrowException()
         }
     }
 
-    public class ExceptionDerivedTests : Exception
+    public class DerivedException : Exception
     {
+        public override string ToString()
+        {
+            return "DerivedException.ToString()";
+        }
+
         [Fact]
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
         public static void Exception_SerializeObjectState()
         {
-            var excp = new ExceptionDerivedTests();
+            var excp = new DerivedException();
             Assert.Throws<PlatformNotSupportedException>(() => excp.SerializeObjectState += (exception, eventArgs) => eventArgs.AddSerializedState(null));
             Assert.Throws<PlatformNotSupportedException>(() => excp.SerializeObjectState -= (exception, eventArgs) => eventArgs.AddSerializedState(null));
         }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public static void Exception_OverriddenToStringOnInnerException()
+        {
+            var inner = new DerivedException();
+            var excp = new Exception("msg", inner);
+
+            Assert.Contains("DerivedException.ToString()", excp.ToString());
```

Comment:
Maybe override Message on DerivedException as well, returning some unique string, and validate that excp.ToString() does not contain what's returned from Message?